nxc/protocols/ssh.py

```diff
@@ -63,124 +79,220 @@ def create_conn_obj(self):
         except socket.error:
             return False
 
-    def client_close(self):
-        self.conn.close()
-
     def check_if_admin(self):
+        self.admin_privs = False
+
+        if self.args.sudo_check:
+            self.check_if_admin_sudo()
+            return
+
         # we could add in another method to check by piping in the password to sudo
         # but that might be too much of an opsec concern - maybe add in a flag to do more checks?
-        stdin, stdout, stderr = self.conn.exec_command("id")
-        if stdout.read().decode("utf-8").find("uid=0(root)") != -1:
-            self.logger.info(f"Determined user is root via `id` command")
-            self.admin_privs = True
-            return True
-        stdin, stdout, stderr = self.conn.exec_command("sudo -ln | grep 'NOPASSWD: ALL'")
-        if stdout.read().decode("utf-8").find("NOPASSWD: ALL") != -1:
-            self.logger.info(f"Determined user is root via `sudo -ln` command")
-            self.admin_privs = True
-            return True
+        self.logger.info("Determined user is root via `id; sudo -ln` command")
+        _, stdout, _ = self.conn.exec_command("id; sudo -ln 2>&1")
+        stdout = stdout.read().decode(self.args.codec, errors="ignore")
+        admin_flag = {
+            "(root)": [True, None],
+            "NOPASSWD: ALL": [True, None],
+            "(ALL : ALL) ALL": [True, None],
+            "(sudo)": [False, f'Current user: "{self.username}" was in "sudo" group, please try "--sudo-check" to check if user can run sudo shell']
+        }
+        for keyword in admin_flag.keys():
+            match = re.findall(re.escape(keyword), stdout)
+            if match:
+                self.logger.info(f'User: "{self.username}" matched keyword: {match[0]}')
+                self.admin_privs = admin_flag[match[0]][0]
+                if not self.admin_privs:
+                    tips = admin_flag[match[0]][1]
+                else:
+                    break
+        if not self.admin_privs and "tips" in locals():
+            self.logger.display(tips)
+        return
+
+    def check_if_admin_sudo(self):
+        if not self.password:
+            self.logger.error("Check admin with sudo not support private key.")
```

Comment:
Just an English fix (and remove the period): "Check admin with sudo does not support using a private key"

Reply:
Fixed!